actors/builtin/verifreg/verified_registry_actor.go

```diff
@@ -318,3 +319,81 @@ func (a Actor) RestoreBytes(rt runtime.Runtime, params *RestoreBytesParams) *abi
 
 	return nil
 }
+
+type RemoveDataCapParams struct {
+	VerifiedClientToRemove addr.Address
+	DataCapAmountToRemove  DataCap
+	VerifierRequest1       RemoveDataCapRequest
+	VerifierRequest2       RemoveDataCapRequest
+}
+
+type RemoveDataCapReturn struct {
+	VerifiedClient addr.Address
+	DataCapRemoved DataCap
+}
+
+// According to unassigned FIP https://github.com/filecoin-project/specs-actors/pull/1546/files, RemoveVerifiedClientDataCap message's sender must be the State.RootKey, and must have two distinct verifiers' signature
+func (a Actor) RemoveVerifiedClientDataCap(rt runtime.Runtime, params *RemoveDataCapParams) *RemoveDataCapReturn {
+
+	// resolve client and verifier addresses in RemoveDataCapParams
+	client, err := builtin.ResolveToIDAddr(rt, params.VerifiedClientToRemove)
+	builtin.RequireNoErr(rt, err, exitcode.ErrIllegalArgument, "failed to resolve client address %s to ID address", params.VerifiedClientToRemove)
+	verifier1, err := builtin.ResolveToIDAddr(rt, params.VerifierRequest1.Verifier)
+	builtin.RequireNoErr(rt, err, exitcode.ErrIllegalArgument, "failed to resolve verifier address %s to ID address", params.VerifierRequest1.Verifier)
+	verifier2, err := builtin.ResolveToIDAddr(rt, params.VerifierRequest2.Verifier)
+	builtin.RequireNoErr(rt, err, exitcode.ErrIllegalArgument, "failed to resolve verifier address %s to ID address", params.VerifierRequest2.Verifier)
+
+	if verifier1 == verifier2 {
+		rt.Abortf(exitcode.ErrIllegalArgument, "need two different verifiers to send remove datacap request got %s and %s that are the same accounts", params.VerifierRequest1.Verifier,
+			params.VerifierRequest2.Verifier)
+	}
+
+	var removedDataCapAmount = params.DataCapAmountToRemove // amount of datacap removed
+	var st State
+
+	rt.StateTransaction(&st, func() {
+		rt.ValidateImmediateCallerIs(st.RootKey)
+		// validate client and verifiers exist
+		verifiedClients, err := adt.AsMap(adt.AsStore(rt), st.Verifiers, builtin.DefaultHamtBitwidth)
```

Comment:
Would be nice for this to become a helper (something like "getVerifiedClientDataCap") that other methods can use